dotnet/src/InternalUtilities/planning/PlannerInstrumentation.cs

```diff
@@ -39,39 +39,62 @@ internal static class PlannerInstrumentation
 
         using var _ = s_activitySource.StartActivity(plannerName);
 
-        if (logger.IsEnabled(LogLevel.Trace))
-        {
-            logger.LogTrace("Plan creation started. Goal: {Goal}", goal); // Sensitive data, logging as trace, disabled by default
-        }
-        else if (logger.IsEnabled(LogLevel.Information))
-        {
-            logger.LogInformation("Plan creation started.");
-        }
+        logger.LogPlanCreationStarted();
+        logger.LogGoal(goal);
 
         TagList tags = new() { { "sk.planner.name", plannerName } };
         long startingTimestamp = Stopwatch.GetTimestamp();
         try
         {
             var plan = await createPlanAsync(planner, kernel, goal, cancellationToken).ConfigureAwait(false);
-
-            if (logger.IsEnabled(LogLevel.Information))
-            {
-                logger.LogInformation("Plan created. Plan:\n{Plan}", planToString(plan));
-            }
+            logger.LogPlanCreatedWithPlan(planToString(plan));
```

Comment:
This is actually a deoptimization. The source generator unfortunately leads folks into a false sense of "it takes care of everything for me", but it doesn't here: the argument `planToString(plan)` will be computed even if the associated log level is disabled. The check for LogLevel.Information should be put back.

Reply:
The `LoggerMessage` attribute has a hardcoded log level. If we put the log level check back, we'd have to make sure the log level in the check matches the one in the `LoggerMessage`. This feels a bit error prone. Let me see if I think of something else.

Reply:
We could consider parameterizing the log level.

Reply:
The PlannerInstrumentation will not longer accept a lambda that convert a plan to a string. A Plan object should override the `ToString` method instead if they want to instrument the planner result.